tests/linked_list_test.c

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+#include <aws/common/linked_list.h>
+#include <aws_test_harness.h>
+
+aws_linked_list_of(int, int_value);
+
+static int test_linked_list_order_prepend_remove() {
+    struct aws_linked_list_node list, *head;
+
+    head = &list;
+    aws_linked_list_init(head);
+
+    struct int_value first = (struct int_value){.value = 1};
+    struct int_value second = (struct int_value){.value = 2};
+    struct int_value third = (struct int_value){.value = 3};
+    struct int_value fourth = (struct int_value){.value = 4};
+
+    aws_linked_list_prepend(head, (struct aws_linked_list_node *)&first);
```

Comment:
This API confuses me, I'm willing to bet in confuses users. I suggest a rename to insert_before and insert_after, and some macros at least for push/pop semantics.